lib/experiments-page.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Bootstraping the Gutenberg experiments page.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * The main entry point for the Gutenberg experiments page.
+ *
+ * @since 5.2.3
+ *
+ * @param string $page The page name the function is being called for, `'gutenberg_customizer'` for the Customizer.
+ */
+function the_gutenberg_experiments( $page = 'gutenberg_page_gutenberg-experiments' ) {
+	?>
+	<div
+		id="experiments-editor"
+		class="blocks-experiments-container"
+	>
+	<?php settings_errors(); ?>
+	<form method="post" action="options.php">
+		<?php settings_fields( 'gutenberg-experiments' ); ?>
+		<?php do_settings_sections( 'gutenberg-experiments' ); ?>           
+		<?php submit_button(); ?>
+	</form>
+	</div>
+	<?php
+}
+
+/**
+ * Set up the experiments settings.
+ *
+ * @since 5.2.3
+ */
+function initialize_experiments_settings() {
+	add_settings_section(
+		'gutenberg_experiments_section',
+		'Experiment settings',
+		'display_experiment_section',
+		'gutenberg-experiments'
+	);
+	add_settings_field(
+		'gutenberg-widgets-screen',
+		'Widgets Screen',
+		'display_experiment_field',
+		'gutenberg-experiments',
+		'gutenberg_experiments_section',
+		array(
+			'Widgets Screen',
+			'gutenberg-widgets-screen',
+		)
+	);
+	add_settings_field(
+		'gutenberg-legacy-widget-block',
+		'Legacy Widget Block',
+		'display_experiment_field',
+		'gutenberg-experiments',
+		'gutenberg_experiments_section',
+		array(
+			'Legacy Widget Block',
+			'gutenberg-legacy-widget-block',
+		)
+	);
+	register_setting(
+		'gutenberg-experiments',
+		'gutenberg-experiments'
+	);
+}
+
+add_action( 'admin_init', 'initialize_experiments_settings' );
+
+/**
+ * Display a checkbox field for a Gutenberg experiment.
+ *
+ * @since 5.2.3
+ *
+ * @param array $args ( $label, $id ).
+ */
+function display_experiment_field( $args ) {
```

Comment:
The `$args` parameter for the `add_settings_field` function is passed exactly as defined with `call_user_func` into the callback therefore it would be nice if $args would be a keyed array such as: 
```
add_settings_field(
	'gutenberg-widgets-screen',
	'Widgets Screen',
	'display_experiment_field',
	'gutenberg-experiments',
	'gutenberg_experiments_section',
	array(
		'label' => 'Widgets Screen',
		'id' => 'gutenberg-widgets-screen',
	)
);
```
because then the code would be easier to read, like:
```
<label for="<?php echo $args['id']; ?>">
	<?php echo $args['label']; ?>
</label>
```
Or you could use list to extract named variables out of the args array, for the same readability goal:
```
function display_experiment_field( $args ) {
	list($label, $id) = $args;
```
but this approach is breakable if the order in the settings definition will change, so I would go for the named keys option.

Reply:
Oooh good tip, thanks Andrei!